tests/test_propagate_updates.py

```diff
@@ -0,0 +1,255 @@
+       The DB only contains the vocabulary id at all times.
+    2. Updating the record would trigger an updated of the referenced fields
+       in ES. Thus, a "re-index" of the touched records would suffice.
+    """
+    expected_lang = [{
+        'id': 'eng',
+        'title': {'en': 'English', 'da': 'Engelsk'}
+    }]
+    # create record with linked entries
+    rec_data = {"metadata": {'title': 'Test', 'languages': [{'id': 'eng'}]}}
+    init_rec = rec_service.create(identity, rec_data)
+    id_ = init_rec.id
+
+    # check record in db
+    from_db = rec_service.read(identity, id_)
+    assert from_db["metadata"]["languages"] == expected_lang
+
+    # check record in es
+    Record.index.refresh()
+    from_es = rec_service.search(identity)
+    assert from_es.total == 1
+    from_es = list(from_es.hits)[0]
+    assert from_es["metadata"]["languages"] == expected_lang
+
+    ###### update lang vocabulary ######
+
+    lang_data["title"]['es'] = "Ingles"
+    up_lang = vocab_service.update(identity, ('languages', 'eng'), lang_data)
+    assert "es" in up_lang.data["title"].keys()
+
+    # check record in db still the same, it is NOT
+    # because in db is only stored as id: [{'id': 'eng'}]
+    # then the RelationsComponent dereferences on read
+    from_db = rec_service.read(identity, id_)
+    assert from_db["metadata"]["languages"][0]["title"] == {
+        'en': 'English', 'da': 'Engelsk', 'es': 'Ingles'
+    }
+
+    # check record in es still the same
+    from_es = rec_service.search(identity)
+    assert from_es.total == 1
+    from_es = list(from_es.hits)[0]
+    assert from_es["metadata"]["languages"][0]["title"] == {
+        'en': 'English', 'da': 'Engelsk'
+    }
+
+    ###### update record title ######
+
+    # this will trigger a re-index of the related fields on ES
+    rec_data["metadata"]["title"] = "Updated"
+    expected_lang = [{
+        'id': 'eng',
+        'title': {'en': 'English', 'da': 'Engelsk', 'es': 'Ingles'}
+    }]
+    updated_rec = rec_service.update(identity, id_, rec_data)
+    updated_rec.data["metadata"]["title"] == "Updated"
+    updated_rec.data["metadata"]["languages"] == expected_lang
+
+    # check record in db
+    from_db = rec_service.read(identity, id_)
+    assert from_db["metadata"]["languages"] == expected_lang
+
+    # check record in es (should be like the old record)
+    Record.index.refresh()
+    from_es = rec_service.search(identity)
+    assert from_es.total == 1
+    from_es = list(from_es.hits)[0]
+    assert from_es["metadata"]["languages"] == expected_lang
+
+
+def test_update_propagation_reindexing(
+    app, identity, rec_service, vocab_service, language_rec, lang_data
+):
+    """Test if reindexing records is not enough.
+
+    The main problem with re-indexing is that the dumper does not get called.
+    Currently, it is called in the Record API cls, we would have to add a call
+    to it in the indexed (makes sense).
```

Comment:
it does call dumps, here https://github.com/inveniosoftware/invenio-indexer/blob/50bb36dda5806eacbf2b3c18fc30ebb4f1f371f5/invenio_indexer/api.py#L384